base/MAPL_Resource.F90

```diff
@@ -0,0 +1,411 @@
+
+      ! No default and not in config, error
+      ! label or default must be present
+      if (.not. label_is_present .and. .not. default_is_present) then
+         if (present(rc)) rc = ESMF_FAILURE
+         return
+      end if
+
+      select type(val)
+      type is(integer(int32))
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(integer(int32))
+               val = default
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, val, label = label, _RC)
+         end if
+      type is(integer(int64))
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(integer(int64))
+               val = default
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, val, label = label, _RC)
+         end if
+      type is(real(real32))
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(real(real32))
+               val = default
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, val, label = label, _RC)
+         end if
+      type is (real(real64))
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(real(real64))
+               val = default
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, val, label = label, _RC)
+         end if
+      type is(character(len=*))
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(character(len=*))
+               val = trim(default)
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, val, label = label, _RC)
+         end if
+      type is(logical)
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(logical)
+               val = default
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, val, label = label, _RC)
+         end if
+         class default
+         _FAIL( "Unupported type")
+      end select
+
+      call ESMF_ConfigGetAttribute(config, printrc, label = 'PRINTRC:', default = 0, _RC)
+
+      ! Can set printrc to negative to not print at all
+      if (MAPL_AM_I_Root() .and. printrc >= 0) then
+         if (label_is_present) then
+            label_to_print = label
+         else
+            label_to_print = trim(label_to_find)
+         end if
+         call print_resource(printrc, label_to_print, val, default=default, _RC)
+      end if
+
+      _RETURN(ESMF_SUCCESS)
+
+   end subroutine MAPL_GetResource_config_scalar
+
+   ! Find value of array variable in config
+   subroutine MAPL_GetResource_config_array(config, compname, vals, label_to_find, default, rc)
```

Comment:
Why are the arguments in this interface in a different order than the scalar case?    And shouldn't compname be optional here as well?   (May need to consult @atrayano and/or @bena-nasa for history)

Reply:
Fixed. Component name is required for the array version but not for the scalar version. So `compname` occurred before `default` which is `optional`. Now it occurs after `compname` and both are `optional`. Functionally, `compname` is still required, so I added an assert that `compname` is present. I added `unusable` before `default` for keyword enforcement.